pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/DeleteAPIHybridClusterIntegrationTest.java

```diff
@@ -0,0 +1,300 @@
+  private String TABLE_NAME;
+  private int nOfflineRows;
+  private int nRealtimeRows;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    super.setUp();
+    TABLE_NAME = super.getTableName();
+    nOfflineRows = numRowsReturned("REALTIME");
+    nRealtimeRows = numRowsReturned("OFFLINE");
+  }
+
+  private int numRowsReturned(String tableType) throws Exception {
+    org.json.JSONObject response = postQuery("select count(*) from '" + TABLE_NAME + "_" + tableType + "'");
+    long start = System.currentTimeMillis();
+    long end = start + 60 * 1000;
+    while (System.currentTimeMillis() < end) {
+      if (!response.get("numDocsScanned").equals(new Integer(0))) {
+        String pinotValue = ((org.json.JSONArray) response.get("aggregationResults")).getJSONObject(0).get("value").toString();
+        return Integer.parseInt(pinotValue);
+      }
+      Thread.sleep(200);
+    }
+    return 0;
+  }
+
+  private boolean checkAllRowsDeleted(String tableType) throws Exception {
+    org.json.JSONObject response = postQuery("select count(*) from '" + TABLE_NAME + "_" + tableType + "'");
+    long start = System.currentTimeMillis();
+    long end = start + 60 * 1000;
+    while (System.currentTimeMillis() < end) {
+      if (response.get("numDocsScanned").equals(new Integer(0))) {
+        return true;
+      }
+      Thread.sleep(200);
+    }
+    return false;
+  }
+
+  private void waitForTableReady() throws Exception {
+    long start = System.currentTimeMillis();
+    long end = start + 60 * 1000;
+    while (System.currentTimeMillis() < end) {
+      int curOfflineRows = numRowsReturned("offline");
+      if (curOfflineRows != nOfflineRows) {
+        Thread.sleep(200);
+      } else {
+        break;
+      }
+    }
+  }
+
+  // @Test TODO: Add back when we use LLC only
+  public void deleteAllRealtimeSegmentsFromGetAPI() throws Exception{
+    sendGetRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forDeleteAllSegmentsWithTypeWithGetAPI(TABLE_NAME, "realtime"));
+
+    checkAllRowsDeleted("REALTIME");
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    Assert.assertEquals(getSegmentsFromJson(postDeleteSegmentList, "realtime").size(), 1);
+    Assert.assertNotSame(getSegmentsFromJson(postDeleteSegmentList, "offline").size(), 0);
+
+    repushRealtimeSegments();
+  }
+
+  private boolean checkOneRowDeleted(String tableType) throws Exception {
+    org.json.JSONObject response = postQuery("select count(*) from '" + TABLE_NAME + "_" + tableType + "'");
+    long start = System.currentTimeMillis();
+    long end = start + 60 * 1000;
+    while (System.currentTimeMillis() < end) {
+      if (tableType.equals("REALTIME")) {
+        if (Integer.parseInt(((org.json.JSONArray) response.get("aggregationResults")).getJSONObject(0).get("value").toString()) != nRealtimeRows) {
```

Comment:
you shoiuld check for nrows being one less than what you had? Or, do you want to rename the method like checkNumRowsReduced()?